internal/planner/max.go

```diff
@@ -0,0 +1,261 @@
+			return nil, err
+		}
+		// If one source property is a float, the result will be a float - no need to check the rest
+		if isTargetFloat {
+			isFloat = true
+			break
+		}
+	}
+
+	return &maxNode{
+		p:                 p,
+		isFloat:           isFloat,
+		aggregateMapping:  field.AggregateTargets,
+		virtualFieldIndex: field.Index,
+		docMapper:         docMapper{field.DocumentMapping},
+	}, nil
+}
+
+func (n *maxNode) Kind() string           { return "maxNode" }
+func (n *maxNode) Init() error            { return n.plan.Init() }
+func (n *maxNode) Start() error           { return n.plan.Start() }
+func (n *maxNode) Spans(spans core.Spans) { n.plan.Spans(spans) }
+func (n *maxNode) Close() error           { return n.plan.Close() }
+func (n *maxNode) Source() planNode       { return n.plan }
+func (n *maxNode) SetPlan(p planNode)     { n.plan = p }
+
+func (n *maxNode) simpleExplain() (map[string]any, error) {
+	sourceExplanations := make([]map[string]any, len(n.aggregateMapping))
+
+	for i, source := range n.aggregateMapping {
+		simpleExplainMap := map[string]any{}
+
+		// Add the filter attribute if it exists.
+		if source.Filter == nil {
+			simpleExplainMap[filterLabel] = nil
+		} else {
+			// get the target aggregate document mapping. Since the filters
+			// are relative to the target aggregate collection (and doc mapper).
+			var targetMap *core.DocumentMapping
+			if source.Index < len(n.documentMapping.ChildMappings) &&
+				n.documentMapping.ChildMappings[source.Index] != nil {
+				targetMap = n.documentMapping.ChildMappings[source.Index]
+			} else {
+				targetMap = n.documentMapping
+			}
+			simpleExplainMap[filterLabel] = source.Filter.ToMap(targetMap)
+		}
+
+		// Add the main field name.
+		simpleExplainMap[fieldNameLabel] = source.Field.Name
+
+		// Add the child field name if it exists.
+		if source.ChildTarget.HasValue {
+			simpleExplainMap[childFieldNameLabel] = source.ChildTarget.Name
+		} else {
+			simpleExplainMap[childFieldNameLabel] = nil
+		}
+
+		sourceExplanations[i] = simpleExplainMap
+	}
+
+	return map[string]any{
+		sourcesLabel: sourceExplanations,
+	}, nil
+}
+
+// Explain method returns a map containing all attributes of this node that
+// are to be explained, subscribes / opts-in this node to be an explainablePlanNode.
+func (n *maxNode) Explain(explainType request.ExplainType) (map[string]any, error) {
+	switch explainType {
+	case request.SimpleExplain:
+		return n.simpleExplain()
+
+	case request.ExecuteExplain:
+		return map[string]any{
+			"iterations": n.execInfo.iterations,
+		}, nil
+
+	default:
+		return nil, ErrUnknownExplainRequestType
+	}
+}
+
+func (n *maxNode) Next() (bool, error) {
+	n.execInfo.iterations++
+
+	hasNext, err := n.plan.Next()
+	if err != nil || !hasNext {
+		return hasNext, err
+	}
+	n.currentValue = n.plan.Value()
+
+	var max *float64
+	for _, source := range n.aggregateMapping {
+		child := n.currentValue.Fields[source.Index]
+		var collectionMax *float64
```

Comment:
todo: This is incorrect for large integers, you cannot convert a very big int to a float and then back again and expect it to have the same value.

suggestion: Add failing tests before fixing this (if you can, you might struggle to set an Int to Int64MaxVal atm via our public clients, as GQL is technically only 32bit Ints)

Reply:
e.g. the below will result in a negative number:
```
a := math.MaxInt64 - 1
		println(fmt.Sprint(a))
		b := float64(a)
		c := int64(b)
		panic(fmt.Sprint(c))
```

Reply:
refactored to use `big.Float`

Reply:
did you try to test this?

Reply:
I've added tests for int max value